From bf25307b6ae78ca6fd3b01277afb56f9e8bb2aaf Mon Sep 17 00:00:00 2001
From: Duncan Bellamy <dunk@denkimushi.com>
Date: Mon, 29 Sep 2025 06:47:58 +0000
Subject: [PATCH] crimson: admin: fix FTBFS with gcc 15

replace stdout and stderr variable names in src/crimson/admin/admin_socket.cc with names that do not clash with the global constant pointers

Fixes: https://tracker.ceph.com/issues/73291

Signed-off-by: Duncan Bellamy <dunk@denkimushi.com>
---
 src/crimson/admin/admin_socket.cc | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/crimson/admin/admin_socket.cc b/src/crimson/admin/admin_socket.cc
index 4bebf5de4e727..36f744fddeef1 100644
--- a/src/crimson/admin/admin_socket.cc
+++ b/src/crimson/admin/admin_socket.cc
@@ -155,14 +155,14 @@ seastar::future<> AdminSocket::execute_line(std::string cmdline,
   LOG_PREFIX(AdminSocket::execute_line);
   INFO("");
   return execute_command({std::move(cmdline)}, {}).then([FNAME, &out, this](auto result) {
-     auto [ret, stderr, stdout] = std::move(result);
+     auto [ret, err_out, out_str] = std::move(result);
      if (ret < 0) {
        ERROR("{}", cpp_strerror(ret));
-       stdout.append(fmt::format("ERROR: {}\n", cpp_strerror(ret)));
-       stdout.append(stderr);
+       out_str.append(fmt::format("ERROR: {}\n", cpp_strerror(ret)));
+       out_str.append(err_out);
      }
      DEBUG("finalizing response");
-     return finalize_response(out, std::move(stdout));
+     return finalize_response(out, std::move(out_str));
   });
 }
 
